mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/query/impl/MongoDBQueryDescriptor.java

```diff
@@ -26,30 +26,47 @@
  * @author Thorsten Möller
  */
 public class MongoDBQueryDescriptor implements Serializable {
-
+	/**
+	 * Enum with operations
+	 * @see <a href="https://docs.mongodb.com/manual/reference/method/js-collection/">list of operations in mongo shell</a>
+	 */
 	public enum Operation {
-		FIND,
-		FINDONE,
-		FINDANDMODIFY,
-		INSERT,
-		REMOVE,
-		UPDATE,
-		COUNT,
+		FIND( false ),
+		FINDONE ( false ),
+		FINDANDMODIFY( false ),
+		INSERT( true ),
+		INSERTONE( false ),
+		INSERTMANY( false ),
+		REMOVE( true ),
+		UPDATE( false ),
+		UPDATEONE( false ),
+		UPDATEMANY( false ),
+		REPLACEONE( false ),
+		COUNT ( false ),
 		/**
 		 * This is used by the query parser when the parsed query requires an aggregation, usually for embedded collections.
 		 */
-		AGGREGATE,
+		AGGREGATE( false ),
 		/**
 		 * This is used for native queries, when the user wants to execute a generic aggregation query.
 		 */
-		AGGREGATE_PIPELINE,
-		DISTINCT;
+		AGGREGATE_PIPELINE( false ),
+		DISTINCT ( false );
+		private boolean deprecated;
+
+		Operation(boolean deprecated) {
+			this.deprecated = deprecated;
+		}
+
+		public boolean isDeprecated() {
```

Comment:
@DavideD May be You right. Fistly, I wants to mark operations that deprecaded in mongodb java 3.x driver. For marking, I upgrade the enum.  You right .... I remove the unused method.